src/sentry/lang/javascript/plugin.py

```diff
@@ -1,36 +1,42 @@
 from __future__ import absolute_import, print_function
 
-from django.conf import settings
 from ua_parser.user_agent_parser import Parse
 
-from sentry.models import Project
 from sentry.plugins import Plugin2
-from sentry.utils import metrics
+from sentry.stacktraces import find_stacktraces_in_data
 
-from .processor import SourceProcessor
+from .processor import JavaScriptStacktraceProcessor
 from .errormapping import rewrite_exception
 
 
 def preprocess_event(data):
-    if settings.SENTRY_SCRAPE_JAVASCRIPT_CONTEXT:
-        project = Project.objects.get_from_cache(
-            id=data['project'],
-        )
+    rewrite_exception(data)
+    fix_culprit(data)
+    inject_device_data(data)
+    generate_modules(data)
+    return data
 
-        allow_scraping = bool(project.get_option('sentry:scrape_javascript', True))
 
-        processor = SourceProcessor(
-            project=project,
-            allow_scraping=allow_scraping,
-        )
-        with metrics.timer('sourcemaps.process', instance=project.id):
-            processor.process(data)
+def generate_modules(data):
+    from sentry.lang.javascript.processor import generate_module
 
-    rewrite_exception(data)
+    for info in find_stacktraces_in_data(data):
+        for frame in info.stacktrace['frames']:
+            platform = frame.get('platform') or data['platform']
+            if platform != 'javascript' or frame.get('module'):
+                continue
+            abs_path = frame.get('abs_path')
+            if abs_path and abs_path.startswith(('http:', 'https:', 'webpack:')):
+                frame['module'] = generate_module(abs_path)
 
-    inject_device_data(data)
 
-    return data
+def fix_culprit(data):
+    exc = data.get('sentry.interfaces.Exception')
+    if not exc:
+        return
+
+    from sentry.event_manager import generate_culprit
+    data['culprit'] = generate_culprit(data)
```

Comment:
Honestly I want to kill this entire thing since this really would have to apply to all processors really.